### ISSUE TITLE
TypeError: Cannot use 'in' operator to search for 'SigningCertUrl'

### ISSUE THREAD

**Issue by c1tt1** [aws/aws-js-sns-message-validator#17]
it looks like there is an inner loop, looping through the values of the keys, in this case the `SigningCertUrl` which is a `.pem` file cannot be looped in that sense which causes this error.

**Comment by jeskew** [comment]
Hi @stern0,

The string `SigningCertUrl` is used as a the left-hand side of an `in` expression [here](https://github.com/aws/aws-js-sns-message-validator/blob/master/index.js#L70), where the right-hand side is expected to be an object (the SNS message) that matches one of the formats described [in the SNS developer guide](https://docs.aws.amazon.com/sns/latest/dg/json-formats.html).

Do you have a bit more detail about what's triggering the `TypeError`? A code sample or a full stack trace would be appreciated.

**Comment by c1tt1** [comment]
What actually triggering that was I had to `JSON.parse` the message, that was the only issue.
This is what I did:

```js
validator.validate(JSON.parse(message), (err, message) => {
      if (err) {
        console.error(err)
        return
      }
      ...
}
```

Maybe that could be implemented internally.

**Comment by c1tt1** [comment]
@jeskew I think the error wasn't that important so would like to close.

**Comment by jeskew** [comment]
@stern0 A fix is included in version 0.3.4
